doc/source/dev/index.rst

```diff
@@ -2,6 +2,31 @@
 Contributing to NumPy
 #####################
 
+Not a coder? Not a problem! NumPy is multi-faceted, and we can use a lot of help.
+These are all activities we'd like to get help with (they're all important, so
+we list them in alphabetical order):
+
+- Code maintenance and development
```

Comment:
Probably this is not the right WIP to contain this remark, 
I think  "Code maintenance and development" could be categorized into:
1. new to open source community
this could contains high level explanation to the project structure, some less sophisticated functions and the logic behind them to give a walkthrough example of converting a requirement into code that satisfies the contribution guide rules.
I think  previous contributors form GSoC could give better input here.   
2. participated into previous open source project

Reply:
Thanks @OmarMerghany. Yes, I agree with your comment. Further down on this page we have the part on how to make code contributions, and that section can be further improved (in another PR probably).